lib/src/block_parser.dart

```diff
@@ -54,10 +54,34 @@ class BlockParser {
   /// The markdown document this parser is parsing.
   final Document document;
 
+  /// The enabled block syntaxes. To turn a series of lines into blocks, each of
+  /// these will be tried in turn. Order matters here.
+  final List<BlockSyntax> blockSyntaxes = [];
+
   /// Index of the current line.
   int _pos;
 
-  BlockParser(this.lines, this.document) : _pos = 0;
+  /// The collection of built-in block parsers.
+  final List<BlockSyntax> standardBlockSyntaxes = const [
+    const EmptyBlockSyntax(),
+    const BlockHtmlSyntax(),
+    const SetextHeaderSyntax(),
+    const HeaderSyntax(),
+    const CodeBlockSyntax(),
+    const BlockquoteSyntax(),
+    const HorizontalRuleSyntax(),
+    const UnorderedListSyntax(),
+    const OrderedListSyntax(),
+    const ParagraphSyntax()
+  ];
+
+  BlockParser(this.lines, this.document) {
+    _pos = 0;
+    if (document.extensions.contains('fenced-code-blocks')) {
+      blockSyntaxes.add(const FencedCodeBlockSyntax());
+    }
+    blockSyntaxes.addAll(standardBlockSyntaxes);
```

Comment:
I agree, I'm not crazy about exposing the innards. It's also unfortunate that the order of the syntaxes matters (maybe not so much with the block-level ones), because then the user can't insert their syntax in the middle.

I'll stick with how they work for InlineSyntaxes. Adding a `blockSyntaxes` field to Document. I can change the API back if we need to later.